SemanticUI/Modules/Dropdown/Selection.elm

```diff
@@ -0,0 +1,399 @@
+
+-}
+scrolling : Bool -> Selection msg option selection -> Selection msg option selection
+scrolling a (Selection config) =
+    Selection { config | scrolling = a }
+
+
+{-| Reduce the space used by a select component.
+A compact selection dropdown has no minimum width.
+A compact button dropdown has reduced padding.
+-}
+compact : Bool -> Selection msg option selection -> Selection msg option selection
+compact a (Selection config) =
+    let
+        variation =
+            case config.variation of
+                Select.Selection sel ->
+                    Select.Selection { sel | compact = a }
+
+                Select.Button but ->
+                    Select.Button (Button.compact a but)
+
+                _ ->
+                    config.variation
+    in
+    Selection { config | variation = variation }
+
+
+{-| Sets a label on the selection control.
+Note that this creates a completely free-form label without any SemanitcUI classes.
+Use `valueLabel` or `valueLabelWithDefault` for labels that reflect the current selection.
+-}
+label : Maybe (Html msg) -> Selection msg option selection -> Selection msg option selection
+label a (Selection config) =
+    Selection { config | labels = Maybe.withDefault [] (Maybe.map List.singleton a) }
+
+
+{-| Generate a label for the current selection.
+If the current selection is not a selected option, return `Err` to generate a default label.
+-}
+valueLabel : (selection -> Result (Html msg) (Html msg)) -> Selection msg option selection -> Selection msg option selection
+valueLabel a ((Selection config) as sel) =
+    let
+        ( isPlaceholder, labelHtml ) =
+            case a config.value of
+                Err html ->
+                    ( True, html )
+
+                Ok html ->
+                    ( False, html )
+    in
+    sel
+        |> label
+            (Just
+                (span
+                    [ classList [ ( "default", isPlaceholder ), ( "text", True ) ]
+                    , -- pointer-events: none ensures that an underlying input will be brought into focus, even with the label on top
+                      -- this is helpful for search dropdowns, though search is not directly implemented as of yet
+                      attribute "pointer-events" "none"
+                    ]
+                    [ labelHtml ]
+                )
+            )
+
+
+{-| Generate a label for the selection values wrapped in `Maybe`.
+Supports the common use-case where no value is selected and a default placeholder label needs to be supplied.
+-}
+valueLabelWithDefault : Html msg -> (selection -> Html msg) -> Selection msg option (Maybe selection) -> Selection msg option (Maybe selection)
+valueLabelWithDefault defaultHtml valueToHtml =
+    valueLabel (Result.fromMaybe defaultHtml << Maybe.map valueToHtml)
+
+
+{-| A dropdown select component with a single active selection.
+-}
+single :
+    { config
+        | value : selection
+        , isSelected : option -> Bool
```

Comment:
This `isSelected` config is probably the one config that I'm the most on the fence about, but I have no better ideas other than possibly:

* Renaming `isSelected` to `optionSelected` to indicate it's specifically on options
* Or changing it to `option -> selection -> Bool` so that one can use `isSelected = (==)` for single selection or `isSelected = List.elem` for multiple selection.